kratos/containers/distributed_sparse_graph.h

```diff
@@ -91,7 +91,7 @@ class DistributedSparseGraph final
       mLocalGraph(LocalSize)
     {
         mNonLocalGraphs.resize(mpComm->Size(),false);
-        mNonLocalLocks.resize(mpComm->Size());
```

Comment:
here i woul also use a spinlock as this happens very very frequently and for a short time